roles/container-engine/cri-o/tasks/main.yaml

```diff
@@ -25,6 +25,20 @@
   tags:
     - facts
 
+- name: disable unified_cgroup_hierarchy in Fedora 31+
+  command: grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
+  when:
+    - ansible_distribution == "Fedora"
+    - (ansible_distribution_major_version | int) >= 31
+    - ansible_proc_cmdline['systemd.unified_cgroup_hierarchy'] is not defined or ansible_proc_cmdline['systemd.unified_cgroup_hierarchy'] != '0'
+
+- name: reboot in Fedora 31+
+  reboot:
+  when:
+    - ansible_distribution == "Fedora"
+    - (ansible_distribution_major_version | int) >= 31
+    - ansible_proc_cmdline['systemd.unified_cgroup_hierarchy'] is not defined or ansible_proc_cmdline['systemd.unified_cgroup_hierarchy'] != '0'
```

Comment:
How about using `block:` for merging `when` conditions into a single one like
```
- block:
  - name: disable unified_cgroup_hierarchy in Fedora 31+
     command: grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
  - name: reboot in Fedora 31+
     reboot:
  when:
    - ansible_distribution == "Fedora"
    - (ansible_distribution_major_version | int) >= 31
    - ansible_proc_cmdline['systemd.unified_cgroup_hierarchy'] is not defined or ansible_proc_cmdline['systemd.unified_cgroup_hierarchy'] != '0'
```
for the readability?

Reply:
I like the idea, I just wonder if should be nice to sync that with [docker](https://github.com/kubernetes-sigs/kubespray/blob/master/roles/container-engine/docker/tasks/main.yml#L50-L62) and [containerd](https://github.com/kubernetes-sigs/kubespray/blob/master/roles/container-engine/docker/tasks/main.yml#L50-L62)

Reply:
Nice point.
To keep the scope of this PR small and easy, It is better to create another PR for the above docker and containerd cases, I feel.

Reply:
We typically avoid blocks and instead use include_tasks module referencing another file in Kubespray. The code in this PR is just fine (2 tasks doesn't really merit a separate file, but 4+ would)